src/index.js

```diff
@@ -46,47 +44,58 @@ module.exports = {
     'id-length': ['error', {
       exceptions: ['_', 'e', 'i']
     }],
+    'id-match': ['error', '^[_a-zA-Z][a-zA-Z0-9]*$|^[A-Z][A-Z]+_[A-Z][A-Z]+$'],
```

Comment:
Was this regex extracted from any particular source? If it's to match camel case or upper case, I believe it can still be improved. For instance, that upper case won't accept numbers, only seems to accept one underscore and requires at least 2 letters at each side of that underscore.

Maybe replace the upper case part of the regex with something like `^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$`.